src/Type/WebMozartAssert/AssertTypeSpecifyingExtension.php

```diff
@@ -462,6 +463,11 @@ private function arrayOrIterable(
 			}
 
 			$specifiedType = TypeCombinator::union(...$newArrayTypes);
+		} elseif ($currentType instanceof GenericObjectType) {
+			$specifiedType = new GenericObjectType(
+				$currentType->getClassName(),
+				[$currentType->getIterableKeyType(), $typeCallback($currentType->getIterableValueType())]
```

Comment:
This is definitely wrong - you have no guarantee that a class has two template types and that it's always TKey + TValue.

Reply:
TBH I'm not really aware of phpstan internals and I'm unsure I'd implement using reasonable amount of effort given this is definitely wrong. Therefore, I'll not pursue this. Thanks for your time tho.